amlip_docs/rst/demo/tensor_inference.rst

```diff
@@ -0,0 +1,144 @@
+.. include:: ../exports/alias.include
+
+.. _demo_inference:
+
+#########################
+TensorFlow Inference Demo
+#########################
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 2
+
+Background
+==========
+
+This demo shows how to implement 2 types of nodes, :ref:`user_manual_nodes_inference` and :ref:`user_manual_nodes_edge`, to perform TensorFlow inference on a given image.
+
+With these 2 nodes implemented, the user can deploy as many nodes of each kind as desired and check the behaviour of a simulated |amlip| network running.
+
+
+Inference refers to the process of using a trained model to make predictions or draw conclusions based on input data.
+It involves applying the learned knowledge and statistical relationships encoded in the model to new, unseen data.
+
+The inference of an image involves passing the image through a trained AI model to obtain a classification based on the learned knowledge and patterns within the model.
+
+TensorFlow
+----------
+
+`TensorFlow <https://www.tensorflow.org/>`_ is an end-to-end machine learning platform with pre-trained models.
+
+Edge Node
+---------
+
+This node simulates an :ref:`user_manual_nodes_edge`.
+It is implemented in |python| using :code:`amlip_py` API.
+
+Inference Node
+--------------
+
+This node simulates an :ref:`user_manual_nodes_inference`.
+It is implemented in |python| using :code:`amlip_py` API.
+To run it, one string argument is required. This will be the model this node will use to make the inference.
+
+Prerequisites
+=============
+
+First of all, check that :code:`amlip_demo_nodes` sub-package is correctly installed.
+If it is not, please refer to :ref:`developer_manual_installation_sources_linux`.
+
+
+The demo requires the following tools to be installed in the system:
+
+.. code-block:: bash
+
+    sudo apt install -y  swig espeak alsa-utils libopencv-dev ffmpeg
+    pip3 install -U pyttsx3 opencv-python pygame
+
+Ensure that you have TensorFlow and TensorFlow Hub installed in your Python environment before proceeding.
+You can install them using pip by executing the following commands:
+
+.. code-block:: bash
+
+    pip3 install tensorflow tensorflow-hub tensorflow-object-detection-api
+
+Additionally, it is required to obtain the TensorFlow model from `TensorFlow Hub <https://tfhub.dev/>`_, follow the steps below:
+
+.. code-block:: bash
+
+    cd ~/AML-IP-demos-ws/src/AML-IP-demos/amlip_demos/resource/tensorflow/models/
+    wget -O centernet_hourglass_512x512_kpts_1.tar.gz https://tfhub.dev/tensorflow/centernet/hourglass_512x512_kpts/1?tf-hub-format=compressed
+    mkdir centernet_hourglass_512x512_kpts_1
+    tar -xvf centernet_hourglass_512x512_kpts_1.tar.gz -C ./centernet_hourglass_512x512_kpts_1
+
+Run demo
+========
+
+The demo that is presented here follows the schema of the figure below:
+
+.. figure:: /rst/figures/tensor_inference_demo.png
```

Comment:
Move this image where I mentioned to add a diagram. This is much cooler and easier to understand.